epan/dissectors/packet-ippusb.c

```diff
@@ -21,70 +21,503 @@
+static gint hf_ippusb_fragment = -1;
+static gint hf_ippusb_fragment_overlap = -1;
+static gint hf_ippusb_fragment_overlap_conflict = -1;
+static gint hf_ippusb_fragment_multiple_tails = -1;
+static gint hf_ippusb_fragment_too_long_fragment = -1;
+static gint hf_ippusb_fragment_error = -1;
+static gint hf_ippusb_fragment_count = -1;
+static gint hf_ippusb_reassembled_in = -1;
+static gint hf_ippusb_reassembled_length = -1;
+static gint hf_ippusb_reassembled_data = -1;
+
+/* Reassemble by default */
+static gboolean global_ippusb_reassemble = TRUE;
+
+static const fragment_items ippusb_frag_items = {
+  &ett_ippusb_fragment,
+  &ett_ippusb_fragments,
+  &hf_ippusb_fragments,
+  &hf_ippusb_fragment,
+  &hf_ippusb_fragment_overlap,
+  &hf_ippusb_fragment_overlap_conflict,
+  &hf_ippusb_fragment_multiple_tails,
+  &hf_ippusb_fragment_too_long_fragment,
+  &hf_ippusb_fragment_error,
+  &hf_ippusb_fragment_count,
+  &hf_ippusb_reassembled_in,
+  &hf_ippusb_reassembled_length,
+  &hf_ippusb_reassembled_data,
+  "IPPUSB fragments"
+};
+
+struct ippusb_multisegment_pdu {
+    guint nxtpdu;
+	guint32 first_frame;
+	guint32 last_frame;
+    guint32 running_size;
+    gboolean finished;
+    gboolean reassembled;
+    gboolean is_chunked;
+    guint32 http_ipp_split;
+
+    guint32 flags;
+    #define MSP_FLAGS_REASSEMBLE_ENTIRE_SEGMENT	0x00000001
+    #define MSP_FLAGS_GOT_ALL_SEGMENTS		0x00000002
+    #define MSP_FLAGS_MISSING_FIRST_SEGMENT     0x00000004
+};
+
+struct ippusb_multisegment_pdu *
+pdu_store(packet_info *pinfo, wmem_tree_t *multisegment_pdus, guint32 previous_first_frame, guint32 previous_is_chunked, guint32 previous_http_ipp_split)
+{
+    struct ippusb_multisegment_pdu *msp;
+
+    msp = wmem_new(wmem_file_scope(), struct ippusb_multisegment_pdu);
+    msp->first_frame = previous_first_frame;
+    msp->last_frame = pinfo->num;
+    msp->finished = FALSE;
+    msp->reassembled = FALSE;
+    msp->is_chunked = previous_is_chunked;
+    msp->http_ipp_split = previous_http_ipp_split;
+    msp->flags = 0;
+    wmem_tree_insert32(multisegment_pdus, pinfo->num, (void *)msp);
+
+    return msp;
+}
+
+struct ippusb_analysis {
+	wmem_tree_t *multisegment_pdus;
+};
+
+static struct ippusb_analysis *
+init_ippusb_conversation_data()
+{
+    struct ippusb_analysis *ippusbd;
+
+    ippusbd = wmem_new0(wmem_file_scope(), struct ippusb_analysis);
+
+    ippusbd->multisegment_pdus=wmem_tree_new(wmem_file_scope());
+
+    return ippusbd;
+}
+
+struct ippusb_analysis *
+get_ippusb_conversation_data(conversation_t *conv, packet_info *pinfo)
+{
+    struct ippusb_analysis *ippusbd;
+
+    if(conv == NULL ) {
+        conv = find_or_create_conversation(pinfo);
+    }
+
+    ippusbd = (struct ippusb_analysis *)conversation_get_proto_data(conv, proto_ippusb);
+
+    if (!ippusbd) {
+        ippusbd = init_ippusb_conversation_data();
+        conversation_add_proto_data(conv, proto_ippusb, ippusbd);
+    }
+
+    if (!ippusbd) {
+      return NULL;
+    }
```

Comment:
Doesn't look like this is needed as ippusbd will be NULL if wmem_new0 fails (e.g. packet-http.c just returns conv_data without this check)